### DISCUSSION TITLE
ESP8266 can't add device in home assistant

### DISCUSSION THREAD

**Discussion by jjtoard** [syssi/esphome-jk-bms#202] [General]
I don't succeed in adding the esp8266 in home assistant. I get warning that I should check that I have an API line in my yaml config but I did add this.
I added 
api:
  encryption:
    key: "my key"

that replaces the config 
mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client

**Comment by syssi** [comment]
1. Is your ESPHome node and HA instance at the same subnet?
2. Can you ping the ESPHome node from the HA instance?
3. You could try to open a telnet session to the IP address of the ESPHome node and port 6053.

I'm pretty sure it's a network issue between your ESPHome node and the HA instance.

**Reply by jjtoard** [reply]
Network is OK,  If i remove everything from uart on in the config, it works without issues.
I also see it in the discovered device list but I can't  add it to homeassistant.  If I remove everything from the uart on, I can add it.  Uploading the full config again, makes that i don't see any sensors added so doesn't look to work.


Tested it with an ESP32 and that works fine but I would like to use the esp8266

**Reply by syssi** [reply]
In this case it is a resource issue. Please remove the half of the sensors (at best the useless) + decrease the global log level to `INFO` and recompile+flash again. I assume the node crashs on startup at the moment.

**Reply by jjtoard** [reply]
Ok thanks,  will test it in a few weeks when I receive the extra jst connectors. At this moment I used the esp32 in my first battery.  I will add an esp in each battery as soon as I have the jst connectors.
Thanks for the feedback and this nice project.  I like the direct integreation in homeassistant above the connection to my victron gx.
